apps/src/templates/teacherDashboard/PrintCertificates.jsx

```diff
@@ -1,26 +1,39 @@
 import React, { Component, PropTypes } from 'react';
 import ProgressButton from '@cdo/apps/templates/progress/ProgressButton';
-import { sectionShape } from './shapes';
 import i18n from '@cdo/locale';
+import $ from 'jquery';
 
 export default class PrintCertificates extends Component {
   static propTypes = {
-    section: sectionShape.isRequired,
+    sectionId: PropTypes.number.isRequired,
     assignmentName: PropTypes.string
   };
 
-  onClickPrintCerts = () => this.certForm.submit();
+  state = {
+    names: [],
+  };
+
+  onClickPrintCerts = () => {
+    $.ajax(`/v2/sections/${this.props.sectionId}/students`).done(result => {
+      const names = result.map(student => student.name);
+      this.setState({names});
+      this.submitForm();
```

Comment:
https://facebook.github.io/react/docs/react-component.html#setstate I think you can give setState a callback, so that it wont call submitForm until completing.